### PR TITLE
Refactor/careers page

### PR DESCRIPTION
**Pull request by yousefalwahami**
## Description

Refactor the careers marketing hub at `/careers`, including new job listings being displayed at `/careers/index` with filters, reusable marketing blocks and assets, also including `careers/[job-slug]` for jobs, no longer using `index/[job-slug]`, Sanity `career/schema` and article handling updates, and marketing route renames (about, research, legal) with nav, footer, and sitemap updates.

## Checklist

- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] Any components that you've modified are accessible.
- [x] You've used [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) where appropriate

### CODE REVIEW

**Comment by vincenttodd** [comment]
visually looks great. was fully prepared for a rickroll on that apply link

**Comment by yousefalwahami** [comment]
> visually looks great. was fully prepared for a rickroll on that apply link

LOL, I missed my chance

File: apps/site/src/messages/careers/en.json [hunk 1]
```diff
@@ -1,10 +1,106 @@
...
+      },
+      "operatingPrinciples": {
+        "heading": "Operating Principles:",
+        "intro": "These principles define how we work together. They are important for establishing our operating culture.",
+        "item1Title": "Find a way.",
+        "item1Body": "We believe in finding a way to do the things that matter. We give agency to individuals and teams to find an approach that works. Ideas come from anywhere, regardless of title or tenure.",
+        "item2Title": "Creativity over control.",
+        "item2Body": "We look to find creative solutions, even if sometimes imperfect, over rigidity and control. Our approach to solving problems is based on best principles (first principles and best practices).",
+        "item3Title": "Update quickly.",
+        "item3Body": "We believe in finding a way to do the things that matter. We give agency to individuals and teams to find an approach that works. Ideas come from anywhere, regardless of title or tenure.",
```
Comment (cristiangramada) [inline apps/site/src/messages/careers/en.json]: The title on line 35 is “Update quickly.” but the paragraph on this line (36) is the same as operatingPrinciples.item1Body (“Find a way.”).

File: apps/site/src/messages/careers/en.json [hunk 2]
```diff
@@ -1,10 +1,106 @@
...
+      "locationFilterAria": "Filter by location",
+      "filterAllTeams": "All teams",
+      "filterAllLocations": "All locations",
+      "locationNotListed": "—",
+      "showingCount": "Showing {visible} jobs of {total}",
+      "noMatches": "No postings match these filters.",
+      "empty": "No postings are listed here right now. See our Todd University externship program on the main careers page."
+    },
     "disclaimers": {
       "0": "Todd is a equal opportunity employer in the regions where we operate.",
```
Comment (cristiangramada) [inline apps/site/src/messages/careers/en.json]: Grammar fix: Todd is `an` equal opportunity...
Change `a` to `an`

File: apps/site/src/app/(unauthenticated)/[locale]/(marketing)/careers/index/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+export async function generateMetadata({
+  params,
+}: {
+  params: Promise<{ locale: string }>;
+}): Promise<Metadata> {
+  const locale = (await params).locale;
+  const t = await getTranslations({ locale, namespace: 'careers.metadata' });
+
+  return {
+    metadataBase: new URL('https://toddagriscience.com'),
```
Comment (cristiangramada) [inline apps/site/src/app/(unauthenticated)/[locale]/(marketing)/careers/index/layout.tsx]: metadataBase is hardcoded to https://toddagriscience.com but openGraph.url uses env.baseUrl. On staging/preview, shared previews and canonical URLs will look wrong or inconsistent. 
Suggest: 
metadataBase: new URL(env.baseUrl) 
to match the environment.

File: apps/site/src/app/(unauthenticated)/[locale]/(marketing)/components/footer/footer.tsx [hunk 4]
```diff
@@ -29,7 +29,7 @@ const Footer = () => {
     { href: '/about', label: t('links.whoWeAre') },
     { href: '/research', label: t('links.whatWeDo') },
     { href: '/news', label: t('links.news') },
-    { href: '/careers', label: t('links.careers') },
+    { href: '/careers/index', label: t('links.careers') },
```
Comment (cristiangramada) [inline apps/site/src/app/(unauthenticated)/[locale]/(marketing)/components/footer/footer.tsx]: Footer Careers goes to /careers/index (job list) instead of /careers (marketing hub). Was that intentional so footer = jobs, or should footer = hub and let the hub route people to listings?

File: apps/site/src/app/(unauthenticated)/[locale]/(marketing)/careers/index/page.tsx
Comment (cristiangramada) [inline apps/site/src/app/(unauthenticated)/[locale]/(marketing)/careers/index/page.tsx:1]: Tests: Other routes in this PR have *.test.tsx; this page doesn’t. 
Add a minimal test: no jobs -> empty-state copy; with jobs -> CareersJobList renders, with Sanity mocked.

File: apps/site/public/robots.txt
Comment (cristiangramada) [inline apps/site/public/robots.txt:1]: Pretty sure this was an accidental formatting change, maybe from Prettier. Unless this was intentional, remove it from the PR.

**Review by cristiangramada — CHANGES_REQUESTED** [review]

File: apps/site/src/messages/careers/en.json (hunk 1 above)
Comment (yousefalwahami) [reply apps/site/src/messages/careers/en.json]: I didn't notice that, @vincenttodd this is the same from the Figma design. Should we remove this line completely?

Comment (vincenttodd) [reply apps/site/src/messages/careers/en.json]: I don't know what any of that is tbh. do whatever needs to be done I guess

Comment (vincenttodd) [reply apps/site/src/messages/careers/en.json]: haha wait now I know what we're talking about. not sure how that happened. For Update quickly:


We come in with a hypothesis and are willing to change our approach as we receive new information. We seek truth and adapt knowing that flexibility is key to progress with agriculture.

Comment (yousefalwahami) [reply apps/site/src/messages/careers/en.json]: perfect! All changes are done.

**Review by cristiangramada — APPROVED** [review]
W
